### PR TITLE
Front-end/Back-end Integration

### LINKED ISSUE #12: IN: Integration the back-end API with the front-end for the Login/Signup pages
- making the API calls from the front-end

### LINKED ISSUE #15: BE: Signup / Login API Routes
- Create a Register route (POST)
    - this will create a new user, return 201
    - validation on the back-end - password should be atleast 6 characters, required fields (email, name), checking confirm and password are the same
    - simple password validation (>6 chars)
    - Save the JWT token in a cookie (should login user as well)
    - Use httpOnly cookies
- Create a Login route (POST)
    - verify username and password
    - return 4xx if things are incorrect
    - Save the JWT token in a cookie
- On successful auth: return token and user object
- User model, don't want to save password directly on the database (hash/salt before we save them)
- Add a basic auth middleware / decorator - we can easily create authenticated routes and get the user object

### PR DESCRIPTION
**Pull request by AltiV**
Implementation of front-end and back-end integration with basic profile page to show persistent login.

### CODE REVIEW

File: client/src/pages/Signup.js [hunk 1]
```diff
@@ -117,31 +155,31 @@ const Signup = () => {
             margin="normal"
             required
             fullWidth
-            name="password1"
+            name="password"
             type="password"
-            id="password1"
+            id="password"
             placeholder="Enter your password"
             onChange={(event) => {
```
Comment (Travmatth) [inline client/src/pages/Signup.js]: just noticed this now, but should just pass `handleInputChange` directly 
```
onChange={handleInputChange}
```

File: client/src/pages/Signup.js [hunk 2]
```diff
@@ -117,31 +155,31 @@ const Signup = () => {
...
             required
             fullWidth
-            name="password2"
+            name="confirmPassword"
             type="password"
-            id="password2"
-            placeholder="Enter your password"
+            id="confirmPassword"
+            placeholder="Confirm your password"
             onChange={(event) => {
```
Comment (Travmatth) [inline client/src/pages/Signup.js]: ditto from line 162

File: client/src/pages/Signup.js (hunk 1 above)
Comment (AltiV) [reply client/src/pages/Signup.js]: Fixed

**Review by sina-jamshidi — APPROVED** [review]
Looks good to me!

**Pull request merged**